backend/dataall/base/utils/naming_convention.py

```diff
@@ -56,6 +56,18 @@ def build_compliant_name(self) -> str:
         suffix = f'-{self.target_uri}' if len(self.target_uri) else ''
         return f"{slugify(self.resource_prefix + '-' + self.target_label[:(max_length - len(self.resource_prefix + self.target_uri))] + suffix, regex_pattern=fr'{regex}', separator=separator, lowercase=True)}"
 
+    def build_compliant_name_with_index(self, index: int = None) -> str:
+        """
+        Builds a compliant AWS resource name
+        """
+        regex = NamingConventionPattern[self.service].value['regex']
+        separator = NamingConventionPattern[self.service].value['separator']
+        max_length = NamingConventionPattern[self.service].value['max_length']
+        index_string = f'-{index}' if index is not None else '- '
```

Comment:
nit: the mix with the conditionals and what is a suffix is a bit confusing. We could improve it by directly:
```
        suffix = f'-{self.target_uri}' if len(self.target_uri) else ''
        index_suff = f'-{index}' if index is not None else ''
        return f"{slugify(self.resource_prefix + '-' + self.target_label[:(max_length - len(self.resource_prefix + self.target_uri + index_string))] + suffix + index_suff, regex_pattern=fr'{regex}', separator=separator, lowercase=True)}"

```